### DISCUSSION TITLE
Parameters passed to the server script were not received

### DISCUSSION THREAD

**Discussion by xxtemps** [ansys/pyansys-geometry#1059] [Q&A] [answered]
![5b2d55d16d4f0956e6b9526cbda2f6f](https://github.com/ansys/pyansys-geometry/assets/12379914/9119138d-f6db-47e7-be9b-ecd6ac1d05a6)
![6e25a25d0693d04a89d1ff220bd4c06](https://github.com/ansys/pyansys-geometry/assets/12379914/8c598cbe-3b43-444e-8661-e11f3887d4f3)
![163f3c62af22bb509f4eca7763e0818](https://github.com/ansys/pyansys-geometry/assets/12379914/224d21e7-27b2-4b19-8e3c-bb48cafd4e17)
What's the problem, please

**Comment by RobPasMue** [comment, marked as answer]
Hi @xxtemps - I believe the main issue is the name of your dictionary inside the SpaceClaim script. It should be "argsDict" and not "argDict".

Nonetheless, you can find more information here: https://geometry.docs.pyansys.com/version/stable/api/ansys/geometry/core/modeler/Modeler.html#Modeler.run_discovery_script_file

**Reply by octoberyy** [reply]
Hi！when I changed "argDict" to "argsDict" in the script, I found that the keyword was not located in the script, and an error "expected float, got str" was reported when running the script through run_discovery_script_file. (My current SCDM version is 24R2, and the ansys-geometry-core version is 0.7.4.).

How can I resolve this issue? I would greatly appreciate your response.

![fig](https://github.com/user-attachments/assets/92c7871e-12ff-4ae3-b69e-412ba541ca50)

**Reply by RobPasMue** [reply]
HI @octoberyy -- thanks for reporting it in this thread. Could you share the PyAnsys Geometry script you are executing with us? I would like to see how the arguments are being passed.

**Reply by RobPasMue** [reply]
Hi @octoberyy -- let's keep the sweep-features related discussion on #1738. You reported here that:

> Hi！when I changed "argDict" to "argsDict" in the script, I found that the keyword was not located in the script, and an error "expected float, got str" was reported when running the script through run_discovery_script_file.

This is the script that I need in order to try and reproduce it. The current script you uploaded does not contain any arguments. Please upload the failing input argument script you mentioned

**Comment by RobPasMue** [comment]
Also, just a suggestion, if you want to import the design back to the Python side, you should store the return value from ``run_discovery_script()``

**Comment by xxtemps** [comment]
Thank you very much. im still learning and exploring

**Reply by RobPasMue** [reply]
Of course! I'm glad to hear that =)

**Comment by xxtemps** [comment]
I think so. Thank you for your advice

**Reply by RobPasMue** [reply]
Anytime!
